ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -171,6 +177,7 @@ public void run() {
 
           for (CompactionInfo ci : potentials) {
             try {
+              Database d = resolveDatabaseAndCache(ci);
```

Comment:
getPoolName here and move `Database d = resolveDatabaseAndCache(ci)` inside of it